### PR TITLE
chore(mesh): eslint

### PR DESCRIPTION
**Pull request by tyc67**
## Notable Changes

- revert #803 
- fix lint error

### CODE REVIEW

File: packages/mesh/next.config.mjs [hunk 1]
```diff
@@ -14,6 +14,9 @@ const nextConfig = {
       },
     ],
   },
+  eslint: {
```
Comment (erase2004) [inline packages/mesh/next.config.mjs:17]: 這應該是多餘的

Comment (tyc67) [reply packages/mesh/next.config.mjs:17]: 因 [next lint](https://nextjs.org/docs/app/building-your-application/configuring/eslint#linting-custom-directories-and-files) 預設只有 `pages/`, `app/`, `components/`, `lib/`, and `src/` directories. 
為了讓 `next lint` 也包含 `constants/`, `context/`, `firebase/`, `hooks/`....等，所以加 `dirs`

Comment (erase2004) [reply packages/mesh/next.config.mjs:17]: 這樣應該明確列舉，而不是整個 package root 都包進去

Comment (tyc67) [reply packages/mesh/next.config.mjs:17]: 修改於 [0cbe5ca](https://github.com/readr-media/Sachiel/pull/820/commits/0cbe5ca65e91de8dbf8a1dc6ad734d124c05f798)

Comment (erase2004) [reply packages/mesh/next.config.mjs:17]: app, component 也需要

Comment (tyc67) [reply packages/mesh/next.config.mjs:17]: sorry, 我搞錯了以為預設就不用加，修改於 [f0d2744](https://github.com/readr-media/Sachiel/pull/820/commits/f0d274483e976a439740fab4da82ed94f06de253)

**Review by erase2004 — APPROVED** [review]
LGTM 👏

File: packages/mesh/app/media/[categorySlug]/_components/category-selector.tsx [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
 import { useRouter } from 'next/navigation'
-import { MouseEventHandler, useEffect, useRef, useState } from 'react'
+import type { MouseEventHandler } from 'react'
+import { useEffect, useRef, useState } from 'react'
 
 import { addCategory, removeCategory } from '@/app/actions/edit-category'
 import Button from '@/components/button'
-import InteractiveIcon, { Icon } from '@/components/interactive-icon'
+import type { Icon } from '@/components/interactive-icon'
+import InteractiveIcon from '@/components/interactive-icon'
```
Comment (caesarWHLee) [inline packages/mesh/app/media/[categorySlug]/_components/category-selector.tsx]: 這邊跟上面的改動應該可以寫成一行用 type 區別就好？ 拆成兩行在撰寫上不太方便

File: packages/mesh/app/story/[id]/_components/api-data-renderer/block-renderer/audio-block.tsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
 'use client'
 
 import Image from 'next/image'
-import { MouseEventHandler, useEffect, useRef, useState } from 'react'
+import type { MouseEventHandler } from 'react'
+import { useEffect, useRef, useState } from 'react'
```
Comment (caesarWHLee) [inline packages/mesh/app/story/[id]/_components/api-data-renderer/block-renderer/audio-block.tsx]: 同上，寫成一行就好

File: packages/mesh/app/story/[id]/_components/story-nav.tsx [hunk 4]
```diff
@@ -4,14 +4,15 @@ import Image from 'next/image'
 import Link from 'next/link'
 import { usePathname } from 'next/navigation'
 
-import Icon, { IconName } from '@/components/icon'
+import type { IconName } from '@/components/icon'
+import Icon from '@/components/icon'
```
Comment (caesarWHLee) [inline packages/mesh/app/story/[id]/_components/story-nav.tsx]: 同上，寫一行就好

File: packages/mesh/components/interactive-icon.tsx [hunk 5]
```diff
@@ -1,4 +1,5 @@
-import Icon, { IconName, Size } from './icon'
+import type { IconName, Size } from './icon'
+import Icon from './icon'
```
Comment (caesarWHLee) [inline packages/mesh/components/interactive-icon.tsx]: 同上，寫一行就好

File: packages/mesh/middleware.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import { cookies } from 'next/headers'
-import { NextRequest, NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
```
Comment (caesarWHLee) [inline packages/mesh/middleware.ts]: 同上，可以改成一行

**Review by caesarWHLee — COMMENTED** [review]
個人覺得 import type 如果要分行寫有點冗贅，如果要改成新的寫法可以 sync 一下大家修改 convention

File: packages/mesh/app/media/[categorySlug]/_components/category-selector.tsx (hunk 2 above)
Comment (tyc67) [reply packages/mesh/app/media/[categorySlug]/_components/category-selector.tsx]: 修改於 [a8375fe](https://github.com/readr-media/Sachiel/pull/820/commits/a8375fe8b7081132af44a6c3b7406cda7da1e498)

File: packages/mesh/app/story/[id]/_components/api-data-renderer/block-renderer/audio-block.tsx (hunk 3 above)
Comment (tyc67) [reply packages/mesh/app/story/[id]/_components/api-data-renderer/block-renderer/audio-block.tsx]: 修改於 [a8375fe](https://github.com/readr-media/Sachiel/pull/820/commits/a8375fe8b7081132af44a6c3b7406cda7da1e498)

File: packages/mesh/app/story/[id]/_components/story-nav.tsx (hunk 4 above)
Comment (tyc67) [reply packages/mesh/app/story/[id]/_components/story-nav.tsx]: 修改於 [a8375fe](https://github.com/readr-media/Sachiel/pull/820/commits/a8375fe8b7081132af44a6c3b7406cda7da1e498)

File: packages/mesh/components/interactive-icon.tsx (hunk 5 above)
Comment (tyc67) [reply packages/mesh/components/interactive-icon.tsx]: 修改於 [a8375fe](https://github.com/readr-media/Sachiel/pull/820/commits/a8375fe8b7081132af44a6c3b7406cda7da1e498)

File: packages/mesh/middleware.ts (hunk 6 above)
Comment (tyc67) [reply packages/mesh/middleware.ts]: 修改於 [a8375fe](https://github.com/readr-media/Sachiel/pull/820/commits/a8375fe8b7081132af44a6c3b7406cda7da1e498)

**Review by caesarWHLee — APPROVED** [review]
再次感謝 🙇

**Pull request merged**